integrations/ScriptLoader.js

```diff
@@ -4,6 +4,7 @@ import logger from "../utils/logUtil";
 const defaultAsyncState = true;
 
 export const LOAD_ORIGIN = "RS_JS_SDK";
+export const IS_NATIVE_SCRIPT = "true";
 
 const ScriptLoader = (id, src, async = defaultAsyncState) => {
```

Comment:
Change this method signature to convert the third argument into an options object.
Also, change `isNative` to use `isRudderSDK` (invert the definition). Default `isRudderSDK` to true as it will minimize the overall changes needed and also prevents any developer errors going forward.


```
options: {
  async: true, // fallback to `defaultAsyncState` if not sent
  isRudderSDK: true // default to true if not sent
}
```

Reply:
Updated the filtering logic.

Reply:
Convert the arguments after `id` and `src` to `options` like I suggested previously.

Reply:
Addressed.